### PR TITLE
[HW_5]: Logging and error handling

### PR DESCRIPTION
**Pull request by Yana-Kalandarova**

### CODE REVIEW

File: hw_3/exceptions/NotFoundError.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export class NotFoundError extends Error {
+    constructor(message) {
+        super();
```
Comment (Ivanko5417) [inline hw_3/exceptions/NotFoundError.js]: I guess you can just pass the message variable to the parent constructor
```javascript
 constructor(message) {
        super(message);
  }      
```

File: hw_3/middleware/validation.js [hunk 2]
```diff
@@ -1,27 +1,25 @@
...
     const userInfo = req.body;
     const { error } = userSchema.validate(userInfo, VALIDATION_OPTIONS);
 
     if (error) {
-        return res.status(400).json(errorResponse(error.details));
+        throw new ValidationError(errorResponse(error.details));
     }
 
-    next();
+    next(error);
```
Comment (Ivanko5417) [inline hw_3/middleware/validation.js]: Looks like you don't need to pass the error to the next function. If the error is not falsy value, the error will be thrown.

**Review by Ivanko5417 — CHANGES_REQUESTED** [review]
Let's add logging of response status and body

**Review by Ivanko5417 — APPROVED** [review]

File: hw_3/exceptions/NotFoundError.js (hunk 1 above)
Comment (Yana-Kalandarova) [reply hw_3/exceptions/NotFoundError.js]: done

File: hw_3/middleware/validation.js (hunk 2 above)
Comment (Yana-Kalandarova) [reply hw_3/middleware/validation.js]: done

**Pull request merged**